vcf-dumper/vcf-dumper-lib/src/main/java/uk/ac/ebi/eva/vcfdump/VariantExporterController.java

```diff
@@ -148,87 +165,66 @@ private void checkParams(List<String> studies, String outputDir, String dbName)
-        if (files != null && files.size() > 0) {
-            query.put(VariantDBAdaptor.FILES, files);
-            if (files.size() == 1) {
-                // this will reduce the data fetched by the mongo driver, improving drastically the performance for databases when many
-                // projects have been previously loaded
-                query.add(VariantDBAdaptor.FILE_ID, files.get(0));
-            }
-        }
-
-        queryParameters.forEach((parameterName, parameterValues) -> {
-            if (VariantDBAdaptor.QueryParams.acceptedValues.contains(parameterName)) {
-                query.add(parameterName, String.join(",", parameterValues));
-            }
-        });
-
-        // exclude fields not needed
-        query.put("exclude", "annotation,sourceEntries.cohortStats");
+//    public QueryOptions getQuery(MultivaluedMap<String, String> queryParameters) {
+//        QueryOptions query = new QueryOptions();
+//        query.put(VariantDBAdaptor.STUDIES, studies);
+//        if (files != null && files.size() > 0) {
+//            query.put(VariantDBAdaptor.FILES, files);
+//            if (files.size() == 1) {
+//                // this will reduce the data fetched by the mongo driver, improving drastically the performance for databases when many
+//                // projects have been previously loaded
+//                query.add(VariantDBAdaptor.FILE_ID, files.get(0));
+//            }
+//        }
+//
+//        queryParameters.forEach((parameterName, parameterValues) -> {
+//            if (VariantDBAdaptor.QueryParams.acceptedValues.contains(parameterName)) {
+//                query.add(parameterName, String.join(",", parameterValues));
+//            }
+//        });
+//
+//        // exclude fields not needed
+//        query.put("exclude", "annotation,sourceEntries.cohortStats");
+//
+//        return query;
+//    }
 
-        return query;
+    public void run() {
+        VCFHeader header = getOutputVcfHeader();
+        VariantContextWriter writer = getWriter();
+        writer.writeHeader(header);
+        writeVariants(writer);
+        writer.close();
     }
 
-    public void run() {
+    public void writeHeader() {
         VCFHeader header = getOutputVcfHeader();
         VariantContextWriter writer = getWriter();
         writer.writeHeader(header);
+        writer.close();
+        logger.info("VCF headers exported");
+    }
 
+    public void writeBlock() {
+        VCFHeader header = getOutputVcfHeader();
+        VariantContextWriter writer = getWriter();
+        writer.setVCFHeader(header);
+        writeVariants(writer);
+        writer.close();
+    }
+
+    private void writeVariants(VariantContextWriter writer) {
         // get all chromosomes in the query or organism, and export the variants for each chromosome
-        Set<String> chromosomes = getChromosomes(query);
+        Set<String> chromosomes = getChromosomes(Collections.singletonList(query.getRegion()));
```

Comment:
what if there are several regions in the query? would this work?

Reply:
Updated to include lists